thorlcr/activities/selfjoin/thselfjoinslave.cpp

```diff
@@ -236,6 +238,10 @@ class SelfJoinSlaveActivity : public CSlaveActivity, public CThorDataLink
         CriticalBlock b(joinHelperCrit);
         rowcount_t p = joinhelper?joinhelper->getLhsProgress():0;
         mb.append(p);
+
+        CRuntimeStatisticCollection mergedStats(spillStats);
+        mergeStats(mergedStats, sorter);
```

Comment:
Hm, I think there is the potential. Only in the case of child queries that re-run though.